modules/post/windows/gather/enum_patches.rb

```diff
@@ -38,34 +42,52 @@ def initialize(info = {})
   end
 
   def run
-    unless load_extapi
-      print_error 'ExtAPI failed to load'
-      return
+    unless session.commands.include?(Rex::Post::Meterpreter::Extensions::Extapi::COMMAND_ID_EXTAPI_WMI_QUERY)
```

Comment:
For normal sessions this is now undefined:

```
msf6 post(windows/gather/enum_patches) > set SESSION 2
SESSION => 2              
msf6 post(windows/gather/enum_patches) > run
                          
[!] SESSION may not be compatible with this module:
[!]  * incompatible session type: shell
[-] Post failed: NameError uninitialized constant Rex::Post::Meterpreter::Extensions::Extapi
[-] Call stack:           
[-]   /home/gwillcox/git/metasploit-framework/modules/post/windows/gather/enum_patches.rb:45:in `run'
[*] Post module execution completed
msf6 post(windows/gather/enum_patches) > 
```

Reply:
Looks like this can be solved with `require 'rex/post/meterpreter/extensions/extapi/command_ids'` which will have minimal impact since it is just including extension definitions. With this the code will work as expected:

```
msf6 post(windows/gather/enum_patches) > reload
[*] Reloading module...   
msf6 post(windows/gather/enum_patches) > run
                          
[!] SESSION may not be compatible with this module:
[!]  * incompatible session type: shell
[-] Post aborted due to failure: no-target: Session does not support Meterpreter ExtAPI WMI queries
[*] Post module execution completed
msf6 post(windows/gather/enum_patches) > 
```

Also seems to work well with Meterpreter:
```
msf6 payload(windows/x64/meterpreter/bind_tcp) > set RHOST 172.28.82.203
RHOST => 172.28.82.203
msf6 payload(windows/x64/meterpreter/bind_tcp) > to_handler
[*] Payload Handler Started as Job 3
msf6 payload(windows/x64/meterpreter/bind_tcp) > 
[*] Started bind TCP handler against 172.28.82.203:4444
[*] Sending stage (200774 bytes) to 172.28.82.203
[*] Meterpreter session 4 opened (172.28.94.235:43743 -> 172.28.82.203:4444) at 2022-09-15 17:41:51 -0500

msf6 payload(windows/x64/meterpreter/bind_tcp) > set SESSION 4
[-] Unknown datastore option: SESSION.
msf6 payload(windows/x64/meterpreter/bind_tcp) > previous
msf6 post(windows/gather/enum_patches) > set SESSION 4 
SESSION => 4
msf6 post(windows/gather/enum_patches) > run

[*] Running module against WIN-BR0CCBA815B (172.28.82.203)

Installed Patches
=================

  HotFix ID  Install Date
  ---------  ------------
  KB5012170  8/18/2022
  KB5016595  8/18/2022
  KB5016693  8/18/2022
  KB5016704  8/18/2022

[*] Patch list saved to /home/gwillcox/.msf4/loot/20220915174215_default_172.28.82.203_enum_patches_414548.txt
[*] Post module execution completed
msf6 post(windows/gather/enum_patches) > 
```